packages/graph/hash_graph/bin/hash_graph/src/main.rs

```diff
@@ -2,255 +2,32 @@
-        {
-            if error.contains::<BaseUriAlreadyExists>() {
-                tracing::info!(%root_account_id, "tried to insert primitive {} data type, but it already exists", title);
-            } else {
-                return Err(error.change_context(GraphError));
-            }
-        } else {
-            tracing::info!(%root_account_id, "inserted the primitive {} data type", title);
-        }
-    }
-
-    // Seed the entity types if they don't already exist
-    let title = link_entity_type.title().to_owned();
-    if let Err(error) = connection
-        .create_entity_type(
-            link_entity_type,
-            OwnedById::new(root_account_id),
-            UpdatedById::new(root_account_id),
-        )
-        .await
-    {
-        if error.contains::<BaseUriAlreadyExists>() {
-            tracing::info!(%root_account_id, "tried to insert {} entity type, but it already exists", title);
-        } else {
-            return Err(error.change_context(GraphError));
-        }
-    } else {
-        tracing::info!(%root_account_id, "inserted the {} entity type", title);
-    }
-
-    Ok(())
-}
+use crate::{args::Args, migrate_database::migrate_database, start_server::start_server};
 
 #[tokio::main]
 async fn main() -> Result<(), GraphError> {
     let args = Args::parse_args();
-    let _log_guard = init_logger(
-        args.log_config.log_format,
-        args.log_config.log_folder,
-        args.log_config.log_level,
-        &args.log_config.log_file_prefix,
-        args.otlp_endpoint.as_deref(),
-    );
-
-    let pool = PostgresStorePool::new(&args.db_info, NoTls)
-        .await
-        .change_context(GraphError)
-        .map_err(|err| {
-            tracing::error!("{err:?}");
-            err
-        })?;
-
-    stop_gap_setup(&pool).await?;
-
-    let rest_router = rest_api_router(
-        Arc::new(pool),
-        DomainValidator::new(args.allowed_url_domain),
-    );
-    let api_address = format!("{}:{}", args.api_host, args.api_port);
-    let addr: SocketAddr = api_address
-        .parse()
-        .into_report()
-        .change_context(GraphError)
-        .attach_printable_lazy(|| api_address.clone())?;
 
-    tracing::info!("Listening on {api_address}");
-    axum::Server::bind(&addr)
-        .serve(rest_router.into_make_service_with_connect_info::<SocketAddr>())
-        .await
-        .expect("failed to start server");
-
-    Ok(())
+    match args.subcommand {
+        args::Subcommand::Server(server_args) => start_server(server_args).await,
+        args::Subcommand::Migrate(migrate_args) => migrate_database(migrate_args).await,
+        args::Subcommand::Completions(CompletionsArgs { shell }) => {
+            clap_complete::generate(
+                shell,
+                &mut Args::augment_args(Command::new(env!("CARGO_PKG_NAME"))),
+                env!("CARGO_PKG_NAME"),
+                &mut std::io::stdout(),
+            );
+            std::process::exit(0);
+        }
```

Comment:
Alongside the above comment: Let's name the files like the subcommand (maybe in a `commands` or `subcommands` folder?). And for consistency, do you want to move the `clap_complete` related things as well?

Reply:
done in d374538afc71d74be00cefec0c7537844375cc2b